### PR TITLE
Prep for adding tags by adding tests

### PR DESCRIPTION
**Pull request by JamesJeffryes**
Added type hinting for existing functions
Added a testsuite
Added an Azure Pipeline config for CI testing

Not bumping setup.py because I'm not planning on releasing this without more updates

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -2,15 +2,15 @@
...
+if python_version() < '3.6':
+    raise ImportError("This module requires Python >=3.6")
 
 with open('README.md', 'r') as in_file:
     long_description = in_file.read()
 
 setup(
     name='clickplc',
-    version='0.2.6',
+    version='0.2.7',
```
Comment (patrickfuller) [inline setup.py:13]: I know we're not really using semantic version as intended (I've always justified as anything < 1.0.0 is free reign) but I think python compatibility changes should be at least a minor version bump. Could also wait until after the tags are added.

File: clickplc/driver.py [hunk 2]
```diff
@@ -73,11 +83,17 @@ async def set(self, address, data):
 
         i = address.index(next(s for s in address if s.isdigit()))
         category, index = address[:i].lower(), int(address[i:])
-        if category not in self.supported:
-            raise ValueError("{} currently unsupported.".format(category))
+        if category not in self.data_types:
+            raise ValueError(f"{category} currently unsupported.")
+        data_type = self.data_types[category].rstrip(digits)
+        for datum in data:
+            if isinstance(datum, int) and data_type == 'float':
```
Comment (patrickfuller) [inline clickplc/driver.py:90]: Your comment on int/bool.

How are we merging this btw? Does this supersede Alex's PR?

File: clickplc/driver.py [hunk 3]
```diff
@@ -122,13 +138,13 @@ async def _get_x(self, start, end):
...
-                output['x{:03d}'.format(current)] = bit
+                output[f'x{current:03}'] = bit
             elif current % 100 == 32:
                 current += 100 - 32
             current += 1
         return output
 
-    async def _get_y(self, start, end):
+    async def _get_y(self, start: int, end: int) -> dict:
         """Read Y addresses. Called by `get`.
```
Comment (patrickfuller) [inline clickplc/driver.py:148]: You haven't done it here but a reminder to leave refactoring x/y to another PR.

File: clickplc/mock.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        elif method == 'write_coils':
+            address, data = args
+            for i, d in enumerate(data):
+                self._coils[address + i] = d
+            return WriteMultipleCoilsResponse(address, data)
+        elif method == 'write_registers':
+            address, data = args
+            for i, d in enumerate(data):
+                self._registers[address + i] = d
+            return WriteMultipleRegistersResponse(address, data)
```
Comment (patrickfuller) [inline clickplc/mock.py:54]: I usually don't like big if blocks but I honestly don't see a better way. The cases are all similar but just different enough to merit this switch-like statement.

**Comment by JamesJeffryes** [comment]
I merged alexes PR into this branch because I wanted access to the tests for his PR and I wanted to be able to add the type checking unit tests in this PR

File: clickplc/driver.py (hunk 2 above)
Comment (alexrudd2) [reply clickplc/driver.py]: my branch got merged into numat/add_tags before this merge

**Comment by alexrudd2** [comment]
@mikeIFTS You may be interested in testing our updates here if you are still using this driver.

**Pull request merged**